packages/nx/src/lock-file/utils/hashing.ts

```diff
@@ -34,29 +34,37 @@ function hashExternalNode(node: ProjectGraphExternalNode, graph: ProjectGraph) {
   if (!graph.dependencies[node.name]) {
     node.data.hash = hashString(hashKey);
   } else {
-    const hashingInput = [hashKey];
-
     // collect all dependencies' hashes
-    traverseExternalNodesDependencies(node.name, graph, hashingInput);
-    node.data.hash = defaultHashing.hashArray(hashingInput.sort());
+    const externalDependencies = traverseExternalNodesDependencies(
+      node.name,
+      graph,
+      new Set([hashKey])
+    );
+    node.data.hash = defaultHashing.hashArray(Array.from(externalDependencies));
```

Comment:
I believe the order is important so that it's consistent? Can we keep the sorting?

Reply:
Thanks! I didn't even see the `.sort()` 😦